mimic/model/heat_objects.py

```diff
@@ -0,0 +1,232 @@
+        """
+        Convenience method for updating action and status.
+        """
+        return self.update_action(action).update_status(status)
+
+    def is_deleted(self):
+        """
+        Checks if stack is in a successfully deleted state.
+        """
+        return self.action == self.DELETE and self.status == self.COMPLETE
+
+    def has_tag(self, tag):
+        """
+        Checks if stack has a tag.
+        """
+        return tag in self.tags
+
+    @classmethod
+    def generate_stack_id(cls):
+        """
+        Generates a stack ID.
+        """
+        return 'test-stack{0}-id-{0}'.format(str(randrange(9999999999)))
+
+    @classmethod
+    def from_creation_request_json(cls, collection, creation_json):
+        """
+        Creates a :obj:`Stack` and adds it to a collection.
+        """
+        def get_tags():
+            tags = creation_json.get('tags', None)
+            return tags.split(',') if tags else []
+
+        stack = cls(
+            stack_name=creation_json['stack_name'],
+            tags=get_tags()
+        )
+        collection.stacks.append(stack)
+        return stack
+
+    def creation_response_json(self):
+        """
+        Returns the response associated with the stack's creation.
+        """
+        return {
+            "stack": {
+                "id": self.stack_id
+            }
+        }
+
+
+stack_creation = EventDescription()
+stack_deletion = EventDescription()
+
+
+@stack_creation.declare_default_behavior
+def default_create_behavior(collection, request, body):
+    """
+    Successfully create a stack.
+    """
+    new_stack = Stack.from_creation_request_json(collection, body)
+    response = new_stack.creation_response_json()
+    request.setResponseCode(201)
+    return json.dumps(response)
+
+
+@stack_deletion.declare_default_behavior
+def default_delete_behavior(collection, request, stack_name, stack_id):
+    """
+    Successfully delete a stack as long as it exists. Updates the stacks status.
+    """
+    stack = collection.stack_by_id(stack_id)
+
+    if not stack:
+        request.setResponseCode(404)
+        return ''
+
+    stack.update_action_and_status(Stack.DELETE, Stack.COMPLETE)
+    request.setResponseCode(204)
+    return ''
+
+
+@attributes(
+    ["tenant_id", "region_name",
+     Attribute("stacks", default_factory=list),
+     Attribute(
+         "behavior_registry_collection",
+         default_factory=lambda: BehaviorRegistryCollection())]
+)
+class RegionalStackCollection(object):
+    """
+    A collection of :obj:`Stack` objects for a region.
+    """
+    def stack_by_id(self, stack_id):
+        """
+        Retrieves a stack by its ID
+        """
+        for stack in self.stacks:
+            if stack.stack_id == stack_id:
+                return stack
+
+    def request_list(self, show_deleted=False, tags=[]):
+        """
+        Tries a stack list operation.
+        """
+        def should_show_stack(stack):
```

Comment:
Done, and added docstring to (hopefully) all method definitions, including the test file.